### PR TITLE
Edit add command parsers

### PR DESCRIPTION
**Pull request by Abhiman2211**
## Description
Adds extra constraints to the add command parsers so that the Alfred recognised when the user has missed one or more field when creating a new entity.

## Checks
- [] No errors when running the tests
- [] Build and checkstyle passes
- [] Run the app and ran 2 commands without failing
- [] Written the baseline test cases for the PR

## Changelog
- []
- []
- []

## Comments for Reviewer (if any)
@hcwong 
@john0227 
@brianyenna 
@Joanna-YJA

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/addcommandparser/AddParticipantCommandParser.java [hunk 1]
```diff
@@ -20,7 +22,7 @@
 /**
  * Parses input arguments and creates a new {@link AddParticipantCommand} object.
  */
-public class AddParticipantCommandParser implements Parser<AddParticipantCommand> {
+public class AddParticipantCommandParser implements Parser<AddCommand> {
```
Comment (john0227) [inline src/main/java/seedu/address/logic/parser/addcommandparser/AddParticipantCommandParser.java]: Why did you change this back?

Comment (Abhiman2211) [reply src/main/java/seedu/address/logic/parser/addcommandparser/AddParticipantCommandParser.java]: I thought we were keeping it consistent: all parsers would take in the AddCommand abstract class you have created their argument. Or actually should I keep the specific command class?

**Comment by Abhiman2211** [comment]
> Btw, is there anything blocking the renaming of AddressBookParser to AlfredParser?

**Comment by Abhiman2211** [comment]
> > Btw, is there anything blocking the renaming of AddressBookParser to AlfredParser?

A lot of the code uses AddressBookParser. But yea I can just refactor and rename it. I'll do it in the next commit

File: src/main/java/seedu/address/logic/parser/addcommandparser/AddParticipantCommandParser.java (hunk 1 above)
Comment (john0227) [reply src/main/java/seedu/address/logic/parser/addcommandparser/AddParticipantCommandParser.java]: I see I see I don't think it matters actually. Think it's okay 👍

Comment (Abhiman2211) [reply src/main/java/seedu/address/logic/parser/addcommandparser/AddParticipantCommandParser.java]: Actually yea it doesn't really matter because in the end AddParticipantCommand is an AddCommand anyways. But I think you would be right in saying that we should make it more specific. I'll change it.
